File: frontend/src/services/adminService.js
// Admin service for API calls
const API_BASE = process.env.REACT_APP_API_BASE || 'http://localhost:8000/api/v1';

export const adminService = {
    // Validation rules management
    async getValidationRules() {
        const response = await fetch(`${API_BASE}/admin/validation-rules`);
        return response.json();
    },

    async createValidationRule(rule) {
        const response = await fetch(`${API_BASE}/admin/validation-rules`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(rule)
        });
        return response.json();
    },

    // API key management
    async getAPIKeys() {
        const response = await fetch(`${API_BASE}/admin/api-keys`);
        return response.json();
    },

    async generateAPIKey(keyData) {
        const response = await fetch(`${API_BASE}/admin/api-keys`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(keyData)
        });
        return response.json();
    },

    // Configuration management
    async getConfig() {
        const response = await fetch(`${API_BASE}/admin/config`);
        return response.json();
    },

    async updateConfig(config) {
        const response = await fetch(`${API_BASE}/admin/config`, {
            method: 'PUT',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(config)
        });
        return response.json();
    }
};

export default adminService;

File: backend/app/services/admin_service.py
"""Admin service for managing enterprise configuration, validation rules, and API keys."""
import uuid
from datetime import datetime
from typing import List, Dict, Any, Optional
from app.models.admin import ValidationRule, ApiKey, EnterpriseConfig

class AdminService:
    """Service class for admin panel operations."""

    @staticmethod
    async def get_enterprise_config() -> Dict[str, Any]:
        """Get current enterprise configuration."""
        # Mock implementation - replace with actual database calls
        return {
            "company_name": "Enterprise Corp",
            "max_calculations_per_day": 10000,
            "enable_advanced_features": True,
            "custom_branding": True,
            "api_rate_limit": 1000,
            "data_retention_days": 90
        }

    @staticmethod
    async def update_enterprise_config(config: EnterpriseConfig) -> Dict[str, Any]:
        """Update enterprise configuration."""
        # Mock implementation - replace with actual database calls
        return {
            "message": "Enterprise configuration updated successfully",
            "config": config.dict()
        }

    @staticmethod
    async def get_validation_rules() -> List[ValidationRule]:
        """Get all validation rules."""
        # Mock implementation - replace with actual database calls
        return [
            ValidationRule(
                id="rule_1",
                name="Age Range Validation",
                field="age",
                rule_type="range",
                parameters={"min": 0, "max": 150},
                is_active=True,
                created_at=datetime.now(),
                updated_at=datetime.now()
            )
        ]

    @staticmethod
    async def create_validation_rule(rule: ValidationRule) -> ValidationRule:
        """Create a new validation rule."""
        rule.id = str(uuid.uuid4())
        rule.created_at = datetime.now()
        rule.updated_at = datetime.now()
        # Mock implementation - replace with actual database calls
        return rule

    @staticmethod
    async def update_validation_rule(rule_id: str, rule: ValidationRule) -> ValidationRule:
        """Update an existing validation rule."""
        rule.id = rule_id
        rule.updated_at = datetime.now()
        # Mock implementation - replace with actual database calls
        return rule

    @staticmethod
    async def delete_validation_rule(rule_id: str) -> None:
        """Delete a validation rule."""
        # Mock implementation - replace with actual database calls
        pass

    @staticmethod
    async def get_api_keys() -> List[ApiKey]:
        """Get all API keys."""
        # Mock implementation - replace with actual database calls
        return [
            ApiKey(
                id="key_1",
                name="Production API Key",
                key_preview="sk_live_****1234",
                permissions=["read", "write"],
                is_active=True,
                created_at=datetime.now(),
                last_used=datetime.now()
            )
        ]

    @staticmethod
    async def generate_api_key(key_data: Dict[str, Any]) -> ApiKey:
        """Generate a new API key."""
        api_key = ApiKey(
            id=str(uuid.uuid4()),
            name=key_data.get("name", "New API Key"),
            key_preview=f"sk_live_****{str(uuid.uuid4())[:4]}",
            permissions=key_data.get("permissions", ["read"]),
            is_active=True,
            created_at=datetime.now(),
            last_used=None
        )
        # Mock implementation - replace with actual database calls
        return api_key

    @staticmethod
    async def revoke_api_key(key_id: str) -> None:
        """Revoke an API key."""
        # Mock implementation - replace with actual database calls
        pass
